Update README with generated table (#64)
nrfconnect/ncs-app-index: README.md

To create an `index.json` of your own, you can either [generate a template](#generating-indexjson) or create a custom file from scratch.
Your custom JSON file must include the following information:

| Property | Requirement | Description |
| -------- | ------------ | ----------- |

> | name | Required | The name of the application repo. Should be the repo-name in the GitHub URL: https://github.com/org/repo-name.|

Reviewer: ```suggestion
| name | Required | The name of the application repository. Should be the `repo-name` in the GitHub URL: https://github.com/org/repo-name.|
```

> | kind | Required | The type of the app repo.|

Reviewer: ```suggestion
| kind | Required | The type of the application repository.|
```

| tags | Required | An array of tags describing the application.|
| releases | Required | The collection of project`s releases.|

> | title | Optional | Human readable name of the repo to be shown in the UI. Defaults to the name property.|

Reviewer: ```suggestion
| title | Optional | Human-readable name of the repository to be shown in the UI. Defaults to the name property.|
```

> | description | Optional | Text describing the application. Inferred from the repo if missing.|

Reviewer: ```suggestion
| description | Optional | Text describing the application. If missing, it is inferred from the repository.|
```

> | manifest | Optional | Alternative filename for the west manifest. Defaults to west.yml.|

Reviewer: ```suggestion
| manifest | Optional | Alternative file name for the west manifest. Defaults to `west.yml`.|
```

> | license | Optional | The name of the application license, e.g. "Apache 2.0". Inferred from the repo if missing.|

Reviewer: ```suggestion
| license | Optional | The name of the application license, for example "Apache 2.0". If missing, it is inferred from the repository.|
```

> | apps | Optional | Glob pattern to find directories containing applications.Applications need a *.conf file and a CMakeLists.txt file at their root. The glob expressions are used to match directories, so no file pattern is necessary.By default, the VS Code extension will assume that there's just a single application sitting at the root of the repo.|

Reviewer: ```suggestion
| apps | Optional | Glob pattern to find directories containing applications. Applications need a `*.conf` file and the `CMakeLists.txt` file at their root. The glob expressions are used to match directories, so no file pattern is necessary. By default, the nRF Connect for Visual Studio Code extension will assume that there is just a single application sitting at the root of the repository.|
```

> | defaultBranch | Optional | The default git branch that the repository is checked out. Inferred from the repo if missing.|

Reviewer: ```suggestion
| defaultBranch | Optional | The default git branch that the repository is checked out at. If missing, it is inferred from the repository.|
```

> | docsUrl | Optional | The URL of the add-on's documentation|

Reviewer: ```suggestion
| docsUrl | Optional | The URL of the add-on's documentation.|
```

> | restricted | Optional | Mark the restricted access to any of the dependencies.|

Reviewer: ```suggestion
| restricted | Optional | Check for marking the restricted access to any of the dependencies.|
```
